notebooks/ch15/bias_cv.md

```diff
@@ -0,0 +1,471 @@
+
+The diagram below illustrates the technique when using five folds:
+
+![feature_5_fold_cv.jpg](https://github.com/DS-100/textbook/blob/master/assets/feature_5_fold_cv.jpg?raw=true)
+
+The biggest advantage of this method is that every data point is used for validation exactly once and for training *k-1* times. Typically, a *k* between 5 to 10 is used, but *k* remains an unfixed parameter. When *k* is small, the error estimate has a lower variance (many validation points) but has a higher bias (fewer training points). Vice versa, with large *k* the error estimate has lower bias but has higher variance. 
+
+$k$-fold cross-validation takes more computation time than the train-validation split since we typically have to refit each model from scratch for each fold. However, it computes a more accurate validation error by averaging multiple errors together for each model.
+
+The `scikit-learn` library provides a convenient [`sklearn.model_selection.KFold`](http://scikit-learn.org/stable/modules/generated/sklearn.model_selection.KFold.html) class to implement $k$-fold cross-validation.
+
+## Bias-Variance Tradeoff
+
+K-fold cross-validation helps us manage the bias-variance tradeoff more accurately. Intuitively, the validation error estimates test error by checking the model's performance on a dataset not used for training; this allows us to estimate both model bias and model variance. K-fold cross-validation also incorporates the fact that the noise in the test set only affects the noise term in the risk equation whereas the noise in the training set only affects bias and model variance in the equation. To choose the final model to use, we select the one that has the lowest validation error.
+
+
+
+## Example: Model Selection for Ice Cream Ratings
+We will use the complete model selection process, including cross-validation, to select a model that predicts ice cream ratings from ice cream sweetness. The complete ice cream dataset and a scatter plot of the overall rating versus ice cream sweetness are shown below.
+
+
+```python
+# HIDDEN
+ice = pd.read_csv('icecream.csv')
+transformer = PolynomialFeatures(degree=2)
+X = transformer.fit_transform(ice[['sweetness']])
+
+clf = LinearRegression(fit_intercept=False).fit(X, ice[['overall']])
+xs = np.linspace(3.5, 12.5, 300).reshape(-1, 1)
+rating_pred = clf.predict(transformer.transform(xs))
+
+#ice.plot.scatter('sweetness', 'overall')
+#plt.plot(xs, rating_pred)
+#plt.title('Degree 2 polynomial fit');
+
+temp = pd.DataFrame(xs, columns = ['sweetness'])
+temp['overall'] = rating_pred
+
+np.random.seed(42)
+x_devs = np.random.normal(scale=0.2, size=len(temp))
+y_devs = np.random.normal(scale=0.2, size=len(temp))
+temp['sweetness'] = np.round(temp['sweetness'] + x_devs, decimals=2)
+temp['overall'] = np.round(temp['overall'] + y_devs, decimals=2)
+
+ice = pd.concat([temp, ice])
+```
+
+
+```python
+ice
```

Comment:
Maybe have a cell with only "ice = pd.read_csv('icecream.csv')" here along with "ice," and do all of the processing afterwards. See below comment as well.